src/Microsoft.AspNet.Mvc.Core/Routing/IUrlHelperFactory.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Mvc.Routing
```

Comment:
This class is fairly visible, not just an implementation detail. Should be in `Microsoft.AspNet.Mvc` with `IUrlHelper`.